index.js

```diff
@@ -1,154 +1,280 @@
 'use strict';
+var vm = require('vm');
 var fs = require('fs');
-var path = require('path');
 var _ = require('lodash');
-var tmpl = require('blueimp-tmpl').tmpl;
 var Promise = require('bluebird');
+var path = require('path');
 Promise.promisifyAll(fs);
 
+var webpack = require('webpack');
+var NodeTemplatePlugin = require('webpack/lib/node/NodeTemplatePlugin');
+var NodeTargetPlugin = require('webpack/lib/node/NodeTargetPlugin');
+var LoaderTargetPlugin = require('webpack/lib/LoaderTargetPlugin');
+var LibraryTemplatePlugin = require('webpack/lib/LibraryTemplatePlugin');
+var SingleEntryPlugin = require('webpack/lib/SingleEntryPlugin');
+
 function HtmlWebpackPlugin(options) {
-  this.options = options || {};
+  // Default options
+  this.options = _.extend({
+    template: __dirname + '/default_index.html',
+    filename: 'index.html',
+    hash: false,
+    inject: true,
+    compile: true,
+    favicon: false,
+    minify: false,
+    chunks: 'all',
+    excludeChunks: [],
+    title: 'Webpack App'
+  }, options);
+  // If the template doesn't use a loader use the blueimp template loader
+  if(this.options.template.indexOf('!') === -1) {
+    this.options.template = require.resolve('./loader.js') + '!' + path.resolve(this.options.template);
+  }
+  // Resolve template path
+  this.options.template = this.options.template.replace(
+    /(\!)([^\/\\][^\!\?]+|[^\/\\!?])($|\?.+$)/,
```

Comment:
What is the purpose of the regexp? I added some extra tests for the loader (will create a PR soon), but the regexp seems to never match anything.